bazel/grpc_build_system.bzl

```diff
@@ -377,6 +469,9 @@ def grpc_cc_binary(name, srcs = [], deps = [], external_deps = [], args = [], da
     copts = []
     if language.upper() == "C":
         copts = ["-std=c99"]
+    if _needs_event_engine_dep(deps):
```

Comment:
Out of curiosity, why is this needed in grpc_cc_binary?  I thought we only care about grpc_cc_test here.

Reply:
Not all targets that indirectly compile `test_config.cc` use cc_test rules, some tests are written as cc_binaries. `test/cpp/end2end:client_crash_test_server` for example. A broad cleanup can maybe be done here separately to eliminate the need.